src/ln/channelmonitor.rs

```diff
@@ -393,6 +393,11 @@ pub struct ChannelMonitor {
 	// the full block_connected).
 	pub(crate) last_block_hash: Sha256dHash,
 	secp_ctx: Secp256k1<secp256k1::All>, //TODO: dedup this a bit...
+
+	//this is all channels transactions that failed in the data_loss_protect protocol
+	//The first index is the height that we attempt to claim it at, this claimed height is by default one week later, and it is there to claim if the other party did not claim
+	transactions_awaiting_claiming : HashMap<u32, Vec<Transaction>>,
```

Comment:
You should be able to just use ChannelMonitor::get_latest_local_commitment_txn(), no need to track new transactions.